### ISSUE TITLE
Edit resources with NULL values

### ISSUE THREAD

**Issue by telmobarros** [SoftwareBrothers/admin-bro#304]
I can't edit resources with NULL values that I desire to keep as NULL.

To begin with I receive this Warning in the browser. (From React I suppose)

> Warning: `value` prop on `input` should not be null. Consider using an empty string to clear the component or `undefined` for uncontrolled components.

Then if I try to save after changing only one value it just clears the form. If I want to update I have to fill all the input fields, even the ones I desire to keep NULL.

**Comment by wojtek-krysiak** [comment]
can you paste a schema and please write which adapter you use (mongoose/typeorm/sequelize)?

**Comment by telmobarros** [comment]
This model for instance, using sequelize adapter. I am able to create a new Partner leaving website as an empty string and its converted to NULL after inserting in the database. But when I try to edit I am not able to do it at all. I tried forcing the allowNull property to true but I get the same error

```
'use strict';
module.exports = (sequelize, DataTypes) => {
	const Partner = sequelize.define('Partner', {
		name: {
			allowNull: false,
			type: DataTypes.STRING(50)
		},
		logo: {
			allowNull: false,
			type: DataTypes.STRING
		},
		website: {
			type: DataTypes.STRING
		}
	}, {});
	return Partner;
};
```

**Comment by mjlawrence83** [comment]
I've seen the #370 fix go in, but it changes all NULLs to be blank string.

I'm planning to have admin-bro working with an existing MSSQL DB via sequelize, and at the moment it will be changing pre-existing records replacing NULLs with ''. I can't afford to allow data changing unexpectedly. Is there a plan to have it handle NULLs or will it continue as its currently implemented? (I can't make any changes via the models I'm using sequelize-auto to generate them on the fly)

**Comment by wojtek-krysiak** [comment]
@mjlawrence83 wow - this is an actual very good solution. We will have to implement this in the core.

**Comment by MichaelLeeHobbs** [comment]
Just sharing the approach I took to resolve this issue.

```
/* jshint indent: 1 */

const NAME = 'Email'

module.exports = (sequelize, DataTypes) => {
    let schema = {
        templateData: {
            type: DataTypes.JSON,
            allowNull: true,
        },
        from: {
            type: DataTypes.STRING,
            allowNull: false,
            unique: false,
            validate: {
                isEmail: true
            }
        },
        sender: {
            type: DataTypes.STRING,
            allowNull: true,
            unique: false,
            validate: {
                isEmail: true
            }
        },
        replyTo: {
            type: DataTypes.STRING,
            allowNull: true,
            unique: false,
            validate: {
                isEmail: true
            }
        },
        priority: {
            type: DataTypes.ENUM('high', 'normal', 'low'),
            allowNull: false,
            unique: false,
            defaultValue: 'normal'
        },
        to: {
            type: DataTypes.TEXT,
            allowNull: false,
            unique: false
        },
        cc: {
            type: DataTypes.TEXT,
            allowNull: true,
            unique: false
        },
        bcc: {
            type: DataTypes.TEXT,
            allowNull: true,
            unique: false
        },
        subject: {
            type: DataTypes.TEXT,
            allowNull: false,
            unique: false
        },
        body: {
            type: DataTypes.TEXT,
            allowNull: false,
            unique: false
        },
        result: {
            type: DataTypes.JSON,
            allowNull: true,
            unique: false
        },
    }
    Object.keys(schema).forEach(key=>{
        if (schema[key].allowNull) {
            let _set = schema[key].set
            schema[key].set = function (val) {
                if (val === '') val = null
                if (_set && 'function' === typeof _set) _set(val)
                else this.setDataValue(key, val)
            }
        }
    })

    let model = sequelize.define(NAME, schema)
    model.associate = function (models) {
        models[NAME].hasOne(models.EmailTemplate)
    }
    model.adminBroConfig = {
        resource: model,
        // options: {
        //     editProperties: ['name'],
        //     listProperties: ['name'],
        //     properties: {
        //         name: {isTitle: true},
        //     }
        // },
    }
    return model
}
```

**Comment by wojtek-krysiak** [comment]
fixed in 3.1.0

**Comment by NoahSaso** [comment]
Is there any documentation about this? I only now know to use `__FORM_VALUE_NULL__` to set properties to null in the db because I stumbled upon this PR.
